rust/connlib/tunnel/src/client.rs

```diff
@@ -284,6 +272,38 @@ where
 
         Ok(())
     }
+
+    fn update_dns_mapping(&mut self) -> bool {
+        let Some(config) = &self.role_state.interface_config else {
+            return false;
+        };
+
+        let effective_dns_servers = effective_dns_servers(
+            config.upstream_dns.clone(),
+            self.role_state.system_resolvers.clone(),
+        );
+
+        if HashSet::<&DnsServer>::from_iter(effective_dns_servers.iter())
+            == HashSet::from_iter(self.role_state.dns_mapping.right_values())
```

Comment:
I don't know how to remember which is left and which is right in the DNS mapping.

Reply:
the types are different so we can't mess it up anyways

Reply:
the right one is the actual servers the left one the sentries